### DISCUSSION TITLE
How to convert a numpy-array (numpy.float64) to a .NET double-array

### DISCUSSION THREAD

**Discussion by MichaelJachan** [pythonnet/pythonnet#1818] [Q&A]
### Environment

-   Pythonnet version: 3.0.0-preview2022-06-03
-   Python version: 3.7
-   Operating System: Win10
-   .NET Runtime: net6.0

Hello! I searched the web and this repository, but i didnt find a standard way to convert a numpy-array (numpy.float64) to a .NET double-array. 

I am coding in C#:

```
double[] csDoubleArray = // some array
PythonEngine.Initialize();
using (Py.GIL()){
  using PyModule pyModule = Py.CreateScope();
  pyModule.Import("numpy");
  pyModule.Set("csDoubleArray ", csDoubleArray .ToPython());
  pyModule.Exec("npDoubleArray = numpy.array(csDoubleArray)");
  pyModule.TryGet("npDoubleArray", out PyObject? pyObject)
  double[] asArray = (double[])pyObject!.AsManagedObject(typeof(double[]))!;// crashes
}
```
//Error: PythonException: 'numpy.float64' value cannot be converted to System.Double

AFAIK, i could write a converter (https://github.com/pythonnet/pythonnet/wiki/Codecs:-customizing-object-marshalling-between-.NET-and-Python)

My Question: But there should already be a standard way, right?

Many thanks, MJ:)

**Comment by lostmsu** [comment]
There's no 'Standard way' because NumPy is but a library, albeit a popular one.

**Reply by MichaelJachan** [reply]
OK, ic. But is there already a code for a converter?

Thank you:)
